Library/Homebrew/diagnostic.rb

```diff
@@ -1111,6 +1112,20 @@ def check_tmpdir
         "TMPDIR #{tmpdir.inspect} doesn't exist." unless tmpdir.nil? || File.directory?(tmpdir)
       end
 
+      def check_tmpdir_executable
+        Tempfile.open("homebrew_check_tmpdir_executable", HOMEBREW_TEMP) do |f|
+          f.write "#!/bin/sh\necho hi\n"
+          f.chmod 0700
+          unless system f.path
+            <<-EOS.undent
+              The directory #{HOMEBREW_TEMP} does not permit executing programs.
+              It is likely mounted "noexec". Please add the following to your ~/.bashrc:
+                export HOMEBREW_TEMP=~/tmp
```

Comment:
On Linuxbrew the user won't likely have write permissions to `/usr/local`, only `/tmp` and her home directory. The situation is different on a Mac.